### PR TITLE
add UpdateFunc to pvcInformer

### LINKED ISSUE #112: controller needs to track pvc updates
Consider:
1. create pvc but forget endpoint anno
2. controller sees pvc but calls `q.forget()`
3. user edits pvc and adds ep anno
4. controller never sees the pvc update.
Therefore the user has to delete the pvc, edit it, then re-create it.

### PR DESCRIPTION
**Pull request by screeley44**
fixes #112

### CODE REVIEW

**Comment by screeley44** [comment]
Question:  Would we ever reprocess a pvc?  I think not, but maybe I'm missing something?

File: pkg/controller/controller_test.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ type operation int
 
 const (
 	opAdd operation = iota
-	opUpdate
+	opUpdate operation = iota
```
Comment (copejon) [inline pkg/controller/controller_test.go]: You only need to declare iota on the first const,  the complier will increment and assign the value to the following constants.

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func (c *Controller) runWorkers() {
 }
 
 // Select pvcs with AnnEndpoint
-// Note: only new pvcs trigger an addition to the work queue. Updated and deleted pvcs
+// Note: only new and updated pvcs potentially trigger an addition to the work queue. and deleted pvcs
```
Comment (jeffvance) [inline pkg/controller/controller.go]: My poorly worded comment. Can you change to: 
```
// Note: only new and updated pvcs potentially trigger an add to the work queue. Deleted pvcs
//    are ignored.
```

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -95,6 +95,13 @@ func (c *Controller) ProcessNextItem() bool {
 		c.queue.Forget(key)
 		return true
 	}
+	if metav1.HasAnnotation(pvc.ObjectMeta, AnnImportPod) {
+		// The pvc may have reached our queue due to a normal update process
+		// however, based on the annotation of an importer pod, we know this was already processed.
+		glog.Infof("processNextItem: annotation %q was found but annotation %q exists indicating this was already processed once, skipping pvc\n", AnnEndpoint, AnnImportPod)
```
Comment (jeffvance) [inline pkg/controller/controller.go]: +1 nice log msg!

File: pkg/controller/controller.go [hunk 4]
```diff
@@ -95,6 +95,13 @@ func (c *Controller) ProcessNextItem() bool {
 		c.queue.Forget(key)
 		return true
 	}
+	if metav1.HasAnnotation(pvc.ObjectMeta, AnnImportPod) {
```
Comment (jeffvance) [inline pkg/controller/controller.go]: Would you mind consolidating all of these  blocks 
```
		c.queue.Forget(key)
		return true
```
to:
```
    return forgetKey(<optional-log-mgs>)
```
where _forgetKey_ looks something like:
```
var forgetkey func(msg string) bool 
     if msg != "" {
        glog.Info(msg)
     }
     c.queue.Forget(key)
     return true
}
```

**Pull request merged**